src/app/search/search.component.ts

```diff
@@ -20,152 +22,105 @@ export class SearchComponent implements OnInit {
   pages: number[];
   noResults: boolean;
   auth: HalDoc;
-  storiesResults: StoryModel[];
-  seriesResults: SeriesModel[];
+  storiesResults: StoryModel[] = [];
+  seriesResults: SeriesModel[] = [];
   loaders: boolean[];
 
   currentPage: number;
   showNumPages: number = 10;
   pagesBegin: number;
   pagesEnd: number;
 
-  searchTextParam: string;
-  searchGenreParam: string;
-  searchSubGenreParam: string;
-  searchSeriesIdParam: number;
-  searchSeriesParam: SeriesModel;
-  searchOrderByParam: string = 'updated_at';
-  searchOrderDescParam: boolean = true;
+  searchStoryParams: SearchStory = new SearchStory();
+  searchSeriesParams: SearchSeries = new SearchSeries();
+
   allSeriesIds: number[];
   allSeries: any;
 
-  orderByOptionsStories: any[] = [
-    {
-      id: 'title',
-      name: 'Story Title'
-    },
-    {
-      id: 'updated_at',
-      name: 'Last Updated'
-    },
-    {
-      id: 'published_at',
-      name: 'When Published'
-    }
-  ];
-
-  orderByOptionsSeries: any[] = [
-    {
-      id: 'title',
-      name: 'Series Title'
-    },
-    {
-      id: 'updated_at',
-      name: 'Last Updated'
-    }
-  ];
+  orderByOptionsStories = SearchStory.ORDERBY_OPTIONS;
+  orderByOptionsSeries = SearchStory.ORDERBY_OPTIONS;
 
   constructor(private cms: CmsService,
               private router: Router,
               private route: ActivatedRoute) {}
 
   ngOnInit() {
-    this.route.params.forEach((params) => {
-      this.selectedTab = params['tab'] || SearchComponent.TAB_STORIES;
-      this.searchSeriesIdParam = +params['seriesId'];
-
-      this.cms.follow('prx:authorization').subscribe((auth) => {
-        this.auth = auth;
-        if (this.selectedTab === SearchComponent.TAB_STORIES) {
-          this.initStorySearch();
-        } else {
-          this.initSeriesSearch();
+    this.cms.follow('prx:authorization').subscribe((auth) => {
+      this.auth = auth;
+
+      this.allSeriesIds = [-1];
+      this.allSeries = {};
+      this.auth.followItems('prx:series', {filters: 'v4'}).subscribe((series) => {
+        for (let s of series) {
+          this.allSeriesIds.push(s.id);
+          this.allSeries[s.id] = new SeriesModel(this.auth, s, false);
         }
+        this.subscribeRouteParams();
```

Comment:
The Observable on the route params parses the route params then calls the search functions which need the auth and/or series to make requests on.